draft-bbs-signatures.md

```diff
@@ -938,22 +966,19 @@ a function that returns the point P in the subgroup G1 corresponding to the cano
 - octets\_to\_point_g2:
 a function that returns the point P in the subgroup G2 corresponding to the canonical representation ostr, or INVALID if ostr is not a valid output of `point_to_octets_g2`.
 
-**Hash to curve parameters**:
+**Generator computation parameters**:
 
 - hash\_to\_curve\_g1:
 A cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve].
 
-- hash\_to\_curve\_g1\_dst: Domain separation tag used in the hash\_to\_curve\_g1 operation.
-
-**Generator Seeds**:
+- generator\_dst: Domain separation tag used in the hash\_to\_curve\_g1 operation.
 
-- message\_generator\_seed: The seed used to generate the message generators which form part of the public parameters used by the BBS signature scheme, Note there are multiple possible scopes for this seed including; a globally shared seed (where the resulting message generators are common across all BBS signatures); a signer specific seed (where the message generators are specific to a signer); signature specific seed (where the message generators are specific per signature). The ciphersuite MUST define this seed OR how to compute it as a pre-cursor operations to any others.
+- generator\_seed: The seed used to compute the generators which form part of the public parameters used by the BBS signature scheme. Note there are multiple possible scopes for this seed including; a globally shared seed (where the resulting message generators are common across all BBS signatures); a signer specific seed (where the message generators are specific to a signer); a signature specific seed (where the message generators are specific per signature). The ciphersuite MUST define this seed OR how to compute it as a pre-cursor operations to any others.
 
-- blind\_value\_generator\_seed: The seed used to calculate the signature blinding value generator (H_s). Similar to the message\_generator\_seed, there are multiple scopes for the blind\_value\_generator\_seed, with the choices being a global seed, a signer specific seed or a signature specific seed. Also, the ciphersuite MUST define this seed OR how to compute it as a pre-cursor operations to any others.
-
-- sig\_domain\_generator\_seed: The seed for calculating the generator used to sign the signature domain separation tag. The scopes and requirements for this seed are the same as the scopes and requirements of the message\_generator\_seed and blind\_value\_generator\_seed.
+- generator\_xof: The extensible-output cryptographic hash function used in generator computation.
```

Comment:
I don't see why we would need to use a different hash function for the generators only?? Since `compute_generators_xof` is optional, it seems weird to me to make the ciphersuite to **have** to define the `generator_xof` parameter (or the other `compute_generators_xof`  specific parameters, like `hash_to_curve_g1`).

Reply:
Yes, more likely it would just use the `hash` function. I'm not sure how best to specify the dependency between the hash used in `hash_to_curve_g1` and the other hash used in this method, and also require that it is an XOF. `hash_to_curve_g1`, `generator_seed` etc are meant to be parameters specific to the `compute_generators_xof` method and should probably be in their own section. I'm not sure yet if there is value in supporting `compute_generators` methods that don't necessarily depend on H2C, but maybe using a method that is agnostic of the hash type is good enough for now.

Reply:
That's a good point! I can't even imagine what those methods (not h2c based) would look like.

> maybe using a method that is agnostic of the hash type is good enough for now.

Yeap! That's my thinking as well. That's why i took a shot in such a procedure [here](https://github.com/decentralized-identity/bbs-signature/pull/161#discussion_r887355556). Another option can be an hkdf based construction (we already have a "optional dependency" on hkdf for keyGen) or a expand_message based one (since 'compute_generators' depends on h2c either way) as i try to propose [here](https://github.com/decentralized-identity/bbs-signature/pull/161#discussion_r887793786). IMO we best use the same hash for everything. We could use a different hash for `hash_to_curve_g1` by just pointing to the defined h2c suite, but for creating the input seed we should use the bbs+ suite defined `hash`.

Another option can be to make the described `compute_generators` procedure a requirement, and define the `generators_xof` parameter to the ciphersuite as required. However, I'm really not a big fun of this approach.